src/AppInstallerCommonCore/AdminSettings.cpp

```diff
@@ -0,0 +1,183 @@
+                return false;
+            }
+        }
+
+        void AdminSettingsInternal::LoadAdminSettings()
+        {
+            auto stream = Settings::GetSettingStream(Settings::Streams::AdminSettings);
+            if (!stream)
+            {
+                AICLI_LOG(Core, Info, << "Admin settings was not found");
+                return;
+            }
+
+            std::string adminSettingsYaml = Utility::ReadEntireStream(*stream);
+
+            YAML::Node document;
+            try
+            {
+                document = YAML::Load(adminSettingsYaml);
+            }
+            catch (const std::exception& e)
+            {
+                AICLI_LOG(YAML, Error, << "Admin settings contained invalid YAML (" << e.what() << ")");
+                return;
+            }
+
+            if (document.IsNull())
+            {
+                AICLI_LOG(Core, Info, << "Admin settings is empty");
+                return;
+            }
+
+            if (!document.IsMap())
+            {
+                AICLI_LOG(Core, Error, << "Admin settings did not contain the expected format");
+                return;
+            }
+
+            TryReadScalar<bool>(document, s_AdminSettingsYaml_LocalManifestFiles, m_settingValues.LocalManifestFiles);
+        }
+
+        void AdminSettingsInternal::SaveAdminSettings() const
+        {
+            YAML::Emitter out;
+            out << YAML::BeginMap;
+            out << YAML::Key << s_AdminSettingsYaml_LocalManifestFiles << YAML::Value << m_settingValues.LocalManifestFiles;
+            out << YAML::EndMap;
+
+            Settings::SetSetting(Settings::Streams::AdminSettings, out.str());
+        }
+    }
+
+    AdminSetting StringToAdminSetting(std::string_view in)
+    {
+        AdminSetting result = AdminSetting::Unknown;
+
+        if (Utility::CaseInsensitiveEquals(s_AdminSettingsYaml_LocalManifestFiles, in))
+        {
+            result = AdminSetting::LocalManifestFiles;
+        }
+
+        return result;
+    }
+
+    std::string_view AdminSettingToString(AdminSetting setting)
+    {
+        switch (setting)
+        {
+        case AdminSetting::LocalManifestFiles:
+            return s_AdminSettingsYaml_LocalManifestFiles;
+        default:
+            return "Unknown"sv;
+        }
+    }
+
+    void EnableAdminSetting(AdminSetting setting)
+    {
+        AdminSettingsInternal adminSettingsInternal;
+        adminSettingsInternal.SetAdminSetting(setting, true);
+    }
+
+    void DisableAdminSetting(AdminSetting setting)
+    {
+        AdminSettingsInternal adminSettingsInternal;
+        adminSettingsInternal.SetAdminSetting(setting, false);
+    }
+
+    bool IsAdminSettingEnabled(AdminSetting setting)
+    {
+        AdminSettingsInternal adminSettingsInternal;
+        bool isEnabled = adminSettingsInternal.GetAdminSettingBoolValue(setting);
+
+        // For some admin settings, even if it's disabled, if the corresponding policy is enabled then override it.
```

Comment:
I would put this above the code that just read in the settings stream and return directly so that we don't waste cycles doing that.